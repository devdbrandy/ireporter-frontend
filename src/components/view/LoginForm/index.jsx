/* eslint-disable react/jsx-no-bind */
import React, { useState } from 'react';
import PropType from 'prop-types';
import PasswordInput from '../Inputs/PasswordInput';

const LoginForm = (props) => {
  const initialState = {
    username: '',
    password: '',
  };
  const [formData, setFormData] = useState(initialState);

  const handleFieldChange = (e) => {
    setFormData({
      ...formData,
      [e.target.name]: e.target.value,
    });
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    const { handleLogin } = props;
    handleLogin(formData);
  };

  return (
    <>
      <form onSubmit={handleSubmit} className="signin--form" id="login-form">
        <div>
          <label htmlFor="username">
            Username
            <div className="input-wrapper">
              <input
                type="text"
                name="username"
                placeholder="Enter username"
                minLength="3"
                required
                onChange={e => handleFieldChange(e)}
              />
            </div>
          </label>
        </div>
        <div>
          <PasswordInput
            name="password"
            placeholder="Enter password"
            handleOnChange={handleFieldChange}
            label="password"
            labelText="Password"
          />
        </div>
        <button type="submit" className="btn btn-primary">Sign In</button>
      </form>
      <div className="signin--links">
        <a href="#!">Forgot Password</a>
      </div>
    </>
  );
};

LoginForm.propTypes = {
  handleLogin: PropType.func.isRequired,
};

export default LoginForm;
