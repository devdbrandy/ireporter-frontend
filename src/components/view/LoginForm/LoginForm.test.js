import React from 'react';
import { shallow } from 'enzyme';
import LoginForm from './index';

describe('<LoginForm />', () => {
  it('renders without crashing given the required props', () => {
    const props = {
      handleLogin: jest.fn()
    };

    const enzymeWrapper = shallow(<LoginForm {...props} />);
    expect(enzymeWrapper).toMatchSnapshot();
  });
});
