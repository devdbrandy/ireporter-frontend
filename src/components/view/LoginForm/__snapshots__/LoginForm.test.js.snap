// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> renders without crashing given the required props 1`] = `
<Fragment>
  <form
    className="signin--form"
    id="login-form"
    onSubmit={[Function]}
  >
    <div>
      <label
        htmlFor="username"
      >
        Username
        <div
          className="input-wrapper"
        >
          <input
            minLength="3"
            name="username"
            onChange={[Function]}
            placeholder="Enter username"
            required={true}
            type="text"
          />
        </div>
      </label>
    </div>
    <div>
      <PasswordInput
        handleOnChange={[Function]}
        label="password"
        labelText="Password"
        minLength={null}
        name="password"
        pattern={null}
        placeholder="Enter password"
        title={null}
      />
    </div>
    <button
      className="btn btn-primary"
      type="submit"
    >
      Sign In
    </button>
  </form>
  <div
    className="signin--links"
  >
    <a
      href="#!"
    >
      Forgot Password
    </a>
  </div>
</Fragment>
`;
